jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/EndPoint.java

```diff
@@ -211,6 +216,24 @@ default int fill(ByteBuffer buffer) throws IOException
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * <p>Receives data into the given buffer from the returned address.</p>
+     * <p>This method should be used to receive UDP data.</p>
+     *
+     * @param buffer the buffer to fill with data
+     * @return the peer address that sent the data, or {@link #EOF}
+     * @throws IOException if the receive fails
+     */
+    default SocketAddress receive(ByteBuffer buffer) throws IOException
```

Comment:
Is this blocking?

Reply:
hmmm few of the existing methods have javadoc making it clear their blocking behaviour.   We should fix this .